test/helpers/contracts/seed/Seed.js

```diff
@@ -120,16 +228,65 @@ class Seed {
       .connect(params.from)
       .approve(this.instance.address, params.fundingAmount);
 
-    await this.instance.connect(params.from).buy(params.fundingAmount);
+    return await this.instance.connect(params.from).buy(params.fundingAmount);
   }
-  async whitelist(params) {
-    if (!params) params = {};
+
+  /**
+   * @param {{from?: SignerWithAddress,
+   *          numberOfRandomClasses?: number,
+   *          classesParameters?: {class1?: {}, class2?: {}, class3?: {}}
+   *        }} params
+   */
+  async addClassAndAllowlist(params = {}) {
+    let functionParams;
     if (!params.from) params.from = await ethers.getSigner(this.admin);
-    if (!params.buyer) params.buyer = (await getNamedTestSigners()).buyer1;
-    if (!params.class) params.class = 0; // Contributor Class
+    if (params.numberOfRandomClasses) {
+      functionParams = await getConvertedParams(
+        types.SEED_ADD_CLASS_AND_WHITELIST_FROM_NUM,
+        params
+      );
+    } else {
+      functionParams = await getConvertedParams(
+        types.SEED_ADD_CLASS_AND_WHITELIST,
+        params
+      );
+    }
+
     await this.instance
       .connect(params.from)
-      .whitelist(params.buyer.address, params.class);
+      .addClassAndAllowlist(...functionParams);
+  }
+
+  /**
+   * @param {{from?: SignerWithAddress, allowlist?: Address[], classes?: number[]}} params
+   */
+  async setAllowlist(params = {}) {
+    if (!params.from) params.from = await ethers.getSigner(this.admin);
+    if (!params.allowlist)
+      params.allowlist = [(await getNamedTestSigners()).buyer1.address];
+    if (!params.classes) params.classes = [0]; // Contributor Class
+
+    await this.instance
+      .connect(params.from)
+      .allowlist(params.allowlist, params.classes);
+  }
+
+  /**
+   * @param {{from?: SignerWithAddress, claimAmount?: string | BigNumber}} params
+   */
+  async claim(params = {}) {
+    if (!params.from) params.from = (await getNamedTestSigners()).buyer1;
+    if (!params.claimAmount)
+      params.claimAmount = await this.calculateClaim(params);
+    return await this.instance.claim(params.from.address, params.claimAmount);
+  }
```

Comment:
(don't have a solution, just mentioning)
I wonder what the overhead will be when introducing a javascript layer on top of the contracts.
Right now it's like
`Testfile -> helpers/Seed.js -> contracts`

In my mind I might prefer
`Testfile + setupParam -> contracts`
Ie. interact with contracts as direct as possible

Reply:
I actually got inspired by the way Balance manages its test. And it is indeed some overhead in writing all the code, but the testing has never been this easy from my experience